src/python/pants/backend/helm/util_rules/tool.py

```diff
@@ -0,0 +1,155 @@
+
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class HelmBinary:
+    path: str
+
+    env: FrozenDict[str, str]
+    immutable_input_digests: FrozenDict[str, Digest]
+
+    def __init__(
+        self,
+        path: str,
+        *,
+        helm_env: Mapping[str, str],
+        local_env: Mapping[str, str],
+        immutable_input_digests: Mapping[str, Digest],
+    ) -> None:
+        self.path = path
+        self.immutable_input_digests = FrozenDict(immutable_input_digests)
+        self.env = FrozenDict({**helm_env, **local_env})
+
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class HelmProcess:
+    argv: tuple[str, ...]
+    input_digest: Digest
+    description: str = dataclasses.field(compare=False)
+    level: LogLevel
+    extra_env: FrozenDict[str, str]
+    extra_immutable_input_digests: FrozenDict[str, Digest]
+    cache_scope: ProcessCacheScope | None
+    output_directories: tuple[str, ...]
+
+    def __init__(
+        self,
+        argv: Iterable[str],
+        *,
+        input_digest: Digest,
+        description: str,
+        level: LogLevel = LogLevel.INFO,
+        output_directories: Iterable[str] | None = None,
+        extra_env: Mapping[str, str] | None = None,
+        extra_immutable_input_digests: Mapping[str, Digest] | None = None,
+        cache_scope: ProcessCacheScope | None = None,
+    ):
+        self.argv = tuple(argv)
+        self.input_digest = input_digest
+        self.description = description
+        self.level = level
+        self.output_directories = tuple(output_directories or ())
+        self.extra_env = FrozenDict(extra_env or {})
+        self.extra_immutable_input_digests = FrozenDict(extra_immutable_input_digests or {})
+        self.cache_scope = cache_scope
+
+
+def _build_helm_env(cache_dir: str, config_dir: str, data_dir: str) -> FrozenDict[str, str]:
+    return FrozenDict(
+        {
+            "HELM_CACHE_HOME": cache_dir,
+            "HELM_CONFIG_HOME": config_dir,
+            "HELM_DATA_HOME": data_dir,
+        }
+    )
+
+
+@rule(desc="Download and configure Helm", level=LogLevel.DEBUG)
+async def setup_helm(helm_subsytem: HelmSubsystem) -> HelmBinary:
+    cache_dir = "__cache"
+    config_dir = "__config"
+    data_dir = "__data"
+
+    downloaded_binary, cache_digest, config_digest, data_digest = await MultiGet(
+        Get(
+            DownloadedExternalTool, ExternalToolRequest, helm_subsytem.get_request(Platform.current)
+        ),
+        Get(Digest, CreateDigest([Directory(cache_dir)])),
+        Get(Digest, CreateDigest([Directory(config_dir)])),
+        Get(Digest, CreateDigest([Directory(data_dir)])),
+    )
+
+    tool_relpath = "__helm"
+    immutable_input_digests = {tool_relpath: downloaded_binary.digest}
+    helm_path = os.path.join(tool_relpath, downloaded_binary.exe)
+    helm_env = _build_helm_env(cache_dir, config_dir, data_dir)
+
+    # TODO Install Global Helm plugins
+    # TODO Configure Helm classic repositories
```

Comment:
Track as Pants issues and put issue numbers here? Fine if no.

Reply:
First TODO would be give support for the `unittest` plugin, probably good that the ticket is something like _Support Helm unittest tests_ instead of just generic Helm plugin support.

The second TODO would be needed to support 3rd party dependencies.

Will create tickets for those.